### ISSUE TITLE
How to Terminate a Crop (e.g., Field Pea) at the Start of Podding in APSIM Classic (v7.10)?

### ISSUE THREAD

**Issue by liluo0423** [APSIMInitiative/APSIMClassic#2250]
Hello APSIM team,

I am currently simulating field pea as a substitute for vetch in APSIM Classic (v7.10), with the goal of using it as a brown manure. I would like to terminate the crop at the start of podding, which is when the crop would typically be cut or killed to incorporate its biomass into the soil.

I understand that APSIM tracks phenological stages, and I assume that the start of podding corresponds to a specific stage in the crop’s growth cycle. My goal is to set a management rule to automatically terminate the field pea crop when it reaches this stage. I would also like to incorporate the remaining biomass into the soil as part of the brown manure process.

Here are my specific questions:

What is the exact phenological stage for field pea at the start of podding in APSIM Classic (v7.10)? How can I identify this from the crop output?

What is the proper syntax for a management script or which management template i should use that terminates the crop when it reaches this stage?

Any guidance on identifying the correct stage and writing the termination rule would be greatly appreciated. Additionally, suggestions on incorporating the terminated biomass into the soil would be helpful.

Thanks in advance!

**Comment by liluo0423** [comment]
Here is how I used in the harvest rule manager 

if ([crop].StageName = 'start_grain_fill' or [crop].plant_status = 'StartPodGrowth') then
              [crop]  end_crop
           endif

and i found it also work if i just
 
if ([crop].StageName = 'StartPodGrowth' ) then
              [crop]  end_crop
           endif

**Comment by peter-devoil** [comment]
Good stuff - glad that is working for you. 

You'll notice that after the end_crop message is sent, the remaining biomass is added to the surface organic matter pool. You can incorporate that into the soil via any of the tillage methods (eg toolbox/Management/Manager (common tasks)/Post Harvest tillage), or simply let it decay.

**Comment by liluo0423** [comment]
> Good stuff - glad that is working for you.
> 
> You'll notice that after the end_crop message is sent, the remaining biomass is added to the surface organic matter pool. You can incorporate that into the soil via any of the tillage methods (eg toolbox/Management/Manager (common tasks)/Post Harvest tillage), or simply let it decay.

Thanks Pete, could you please elaborate a little more on how to incorporate that into the soil? What we think of is using it as brown manure, and I guess the option here would be to let it decay? Appreciate your response.

**Comment by peter-devoil** [comment]
The SOM module is independant of the SoilN module that maintains 3 C&N pools - fom, biom and hum. When SOM decays  biomass (a daily function of temperature, surface moisture & C:N ratio) it is removed from SOM and added to surface fom (in soiln). That's the normal behaviour in apsim - ie there is no management required to make this happen.

To incorporate it by plough you send the SOM module a message like
```
'[surfaceommodule]' tillage type = 'disc'
```
that is in logic component, eg:
![image](https://github.com/user-attachments/assets/92b96379-86fa-46a6-81b8-4076b8803293)

The SOM module will look up the depth and fraction associated with a "disc", and add biomass to the fom pool to the depth specified.

In short, if you want to simulate surface mulch, then to dothing.. Be careful that obscene amounts (eg from sugar cane) dont develop over time. C:N ratio of mulch is important for correct decomposition rates.

**Comment by liluo0423** [comment]
> The SOM module is independant of the SoilN module that maintains 3 C&N pools - fom, biom and hum. When SOM decays biomass (a daily function of temperature, surface moisture & C:N ratio) it is removed from SOM and added to surface fom (in soiln). That's the normal behaviour in apsim - ie there is no management required to make this happen.
> 
> To incorporate it by plough you send the SOM module a message like
> 
> ```
> '[surfaceommodule]' tillage type = 'disc'
> ```
> 
> that is in logic component, eg: ![image](https://private-user-images.githubusercontent.com/1924386/371908996-92b96379-86fa-46a6-81b8-4076b8803293.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3Mjc2NTcwMzYsIm5iZiI6MTcyNzY1NjczNiwicGF0aCI6Ii8xOTI0Mzg2LzM3MTkwODk5Ni05MmI5NjM3OS04NmZhLTQ2YTYtODFiOC00MDc2Yjg4MDMyOTMucG5nP1gtQW16LUFsZ29yaXRobT1BV1M0LUhNQUMtU0hBMjU2JlgtQW16LUNyZWRlbnRpYWw9QUtJQVZDT0RZTFNBNTNQUUs0WkElMkYyMDI0MDkzMCUyRnVzLWVhc3QtMSUyRnMzJTJGYXdzNF9yZXF1ZXN0JlgtQW16LURhdGU9MjAyNDA5MzBUMDAzODU2WiZYLUFtei1FeHBpcmVzPTMwMCZYLUFtei1TaWduYXR1cmU9ZGRjOGM3NjVkYTkzODkyMjc4YjFjYzUwOWNjOWFjZWJlNjVkYTkyYjg1YmRiYmIzNmUzZTRiOTYwYjM3OGViNiZYLUFtei1TaWduZWRIZWFkZXJzPWhvc3QifQ.MDOQIHPVxH0YcVe93xnQI5d7xLa46h7bAPidbf8wixc)
> 
> The SOM module will look up the depth and fraction associated with a "disc", and add biomass to the fom pool to the depth specified.
> 
> In short, if you want to simulate surface mulch, then to dothing.. Be careful that obscene amounts (eg from sugar cane) dont develop over time. C:N ratio of mulch is important for correct decomposition rates.

Thanks Pete, your explaination is very clear!
